homeassistant/components/mqtt/mixins.py

```diff
@@ -1144,11 +1144,6 @@ def _set_entity_name(self, config: ConfigType) -> None:
                 )
             elif (device_name := config[CONF_DEVICE][CONF_NAME]) == entity_name:
                 self._attr_name = None
-                self._issue_key = (
-                    "entity_name_is_device_name_discovery"
-                    if self._discovery
-                    else "entity_name_is_device_name_yaml"
-                )
```

Comment:
This will also remove the issue for yaml configured entities

Reply:
🤦

Reply:
Oh! Missed this message.

Soft re-setted this branch, fix the error and pushed out a new version of the PR.